src/extensions/agent_modules/shared/upd_utils/src/adu_upd_utils.c

```diff
@@ -127,4 +128,18 @@ void AduUpdUtils_TransitionState(ADU_UPD_STATE newState, ADUC_Update_Request_Ope
         newState,
         AduUpdState_str(newState));
     updateData->updState = newState;
+
+    if (newState == ADU_UPD_STATE_IDLEWAIT)
+    {
+        srand(time(NULL));
+        int jitter_seconds = (rand() % 121) + 60; // between 1 and 3 minutes of jitter
+        time_t now = ADUC_GetTimeSinceEpochInSeconds();
+        context->nextExecutionTime = now + context->pollingIntervalInSeconds + jitter_seconds;
```

Comment:
We need toc heck if context is null before assigning/accessing variables unless we want the boom to do the error checking for us.

Reply:
I would actually prefer it crashes so it's discoverable and can get a core dump as it should never be NULL here.